ui/apps/platform/cypress/integration/violations/violations.test.js

```diff
@@ -21,22 +21,38 @@ describe('Violations page', () => {
     it('should select item in left nav', () => {
         visitViolations();
 
-        cy.get(ViolationsPageSelectors.navLink).should('have.class', 'pf-m-current');
+        cy.get(selectors.navLink).should('have.class', 'pf-m-current');
     });
 
     it('should have violations in table', () => {
         visitViolationsWithFixture('alerts/alerts.json');
 
         const count = 2;
-        cy.get(ViolationsPageSelectors.resultsFoundHeader(count));
-        cy.get(ViolationsPageSelectors.table.rows).should('have.length', count);
+        cy.get(selectors.resultsFoundHeader(count));
+        cy.get(selectors.table.rows).should('have.length', count);
     });
 
-    it('should have Lifecycle column in table', () => {
+    it('should have columns in table', () => {
         visitViolationsWithFixture('alerts/alerts.json');
 
-        cy.get(ViolationsPageSelectors.table.column.lifecycle);
-        cy.get(`${ViolationsPageSelectors.firstTableRow} td[data-label="Lifecycle"]`).should(
+        cy.get('th[scope="col"]:contains("Policy")');
+        cy.get('th[scope="col"]:contains("Entity")');
+        cy.get('th[scope="col"]:contains("Type")');
+        cy.get('th[scope="col"]:contains("Enforced")');
+        cy.get('th[scope="col"]:contains("Severity")');
+        cy.get('th[scope="col"]:contains("Categories")');
+        cy.get('th[scope="col"]:contains("Lifecycle")');
+        cy.get('th[scope="col"]:contains("Time")');
```

Comment:
nit: This doesn't validate the order of the columns.
(Both plus and minus: product desire to change presentation order of columns won't break tests, but tests also can't validate that change or any other regression.)

Overall, fine to leave as-is here, but worth deciding as a team if we want to move toward that tighter testing style.

Reply:
Indeed, a good point to discuss and decide.

Reply:
Here is rough draft of a pattern to easily change order of columns without need to edit index:

```js
[
    'Policy',
    'Entity',
    'Type',
    // and so on
].forEach((columnText, index) => {
    cy.get(`th[scope="col"]:nth(${index}):contains("${columnText}")`);
});
```